### PR TITLE
feat: toggle sunset mode

### LINKED ISSUE #118: Мodified Script Sunset.sh, adding a widget in Waybar
I was modified by the script for working with Wlsunset.
Added features:
1. Widget Waybar, allowing you to turn on and disable "night mode", as well as visualize the current state.
2. Configuration file, and auto creation mechanism
3. Ability to not use location
4. Checking the correctness of the answer from FreeGeoip.App

P.S. I do not have experience in participating in such big projects as Manjaro. Is I doing correctly by sending trouble through "Issues"? 

Changed files:
`~/.config/waybar/config`
Do not forget to register this seen in Style.css and in the section "Modules-Right" 
~~~json
    "custom/sunset": {
	"interval": 2,
        "tooltip": false,
        "return-type": "json",
        "format" : "{icon}",
        "format-icons": {
            "on": "ﯦ" ,
            "off": "ﯧ" 
        },
        "exec": "/bin/sh /usr/share/sway/scripts/sunset.sh",
	"on-click": "swaymsg exec /usr/share/sway/scripts/sunset.sh 'toggle'",
    },
~~~

`/usr/share/sway/scripts/sunset.sh`
~~~bash
#!/bin/sh
#Checks if a configuration file exists, if necessary creates 
function checkConfig(){
	if [ ! -f "$HOME/.config/wlsunset/config" ];
	then 
		mkdir -p $HOME/.config/wlsunset
echo 'temp_low="4000"
temp_high="6500"
sunrise="07:00"
sunset="23:00"
duration="900"
location="off"
' >> $HOME/.config/wlsunset/config
	fi
}

#Startup function
function onSunSet(){
	checkConfig
	source $HOME/.config/wlsunset/config

	if [ ${location} = "on" ]; 
	then
		CONTENT=$(curl -s https://freegeoip.app/json/)
		longitude=$(echo $CONTENT | jq .longitude)
		latitude=$(echo $CONTENT | jq .latitude)
		if [ -e $longitude ];
		then
			echo location ERROR: freegeoip.app
			longitude='65'
			latitude='60'
		else
			echo location OK:  $latitude $longitude
		fi	
		wlsunset -l $latitude -L $longitude -t $temp_low -T $temp_high -d $duration &
	else
		wlsunset -t $temp_low -T $temp_high  -d $duration -S $sunrise -s $sunset &
	fi

}

#Accepts managing parameter 
case $1'' in
	'off')
   	pkill wlsunset
	;;

	'on')
	onSunSet
	;;

	'toggle')
   	if pkill -0 wlsunset
	then
		pkill wlsunset
	else
		onSunSet
	fi
    ;;
esac

#Returns a string for Waybar 
if pkill -0 wlsunset
then
	class="on"
else
	class="off"
fi	

printf '{"alt":"%s"}\n' "$class"
~~~

You should also change the file `/etc/sway/config.d/99-autostart-applications.conf`
~~~bash
'[ -x "$(command -v wlsunset)" ] && $HOME/.config/sway/scripts/sunset.sh "on"'
~~~

### PR DESCRIPTION
**Pull request by boredland**
closes https://github.com/Manjaro-Sway/manjaro-sway/issues/118

### CODE REVIEW

**Comment by Mathew-D** [comment]
The changes look good but one question, what if someone doesn't have sunset installed?  The waybar module would still be setup and load.   You might want to add something to the FAQ about this so people know to remove it from waybar if they don't have sunset.

**Comment by boredland** [comment]
> what if someone doesn't have sunset installed?

perhaps we could instead add an "exec-if" that checks if the binary is there?

**Comment by boredland** [comment]
@Mathew-D I added an exec-if and it seems to work like a charm - tyvm

**Comment by Mathew-D** [comment]
I don't run sunset but I made all the changes and it doesn't show on the waybar so the exec-if works. Now I noticed we need a config file now which I believe is to customize it better now correct?

**Comment by boredland** [comment]
> I don't run sunset but I made all the changes and it doesn't show on the waybar so the exec-if works. Now I noticed we need a config file now which I believe is to customize it better now correct?

yes, it's from the issue @MuratovAS created. As I understand it, it fullfills four jobs:

1. configure a location to not rely on geoip to work
2. fall back to defaults if geoip not available
3. being able to customize personal on/off times (when you don't want to use sunrise-data for your coordinates)
4. being able to customize color temp

File: community/sway/usr/share/sway/scripts/sunset.sh [hunk 1]
```diff
@@ -1,5 +1,58 @@
...
+		then
+			echo location ERROR: freegeoip.app
+			longitude='65'
+			latitude='60'
+		else
+			echo location OK:  $latitude $longitude
+		fi	
+		wlsunset -l $latitude -L $longitude -t $temp_low -T $temp_high -d $duration &
+	else
+		wlsunset -t $temp_low -T $temp_high  -d $duration -S $sunrise -s $sunset &
```
Comment (boredland) [inline community/sway/usr/share/sway/scripts/sunset.sh]:
```suggestion
		wlsunset -t $temp_low -T $temp_high -d $duration -S $sunrise -s $sunset &
```

File: community/sway/usr/share/sway/scripts/sunset.sh [hunk 2]
```diff
@@ -1,5 +1,58 @@
...
+
+	if [ ${location} = "on" ]; 
+	then
+		CONTENT=$(curl -s https://freegeoip.app/json/)
+		longitude=$(echo $CONTENT | jq .longitude)
+		latitude=$(echo $CONTENT | jq .latitude)
+		if [ -e $longitude ];
+		then
+			echo location ERROR: freegeoip.app
+			longitude='65'
```
Comment (boredland) [inline community/sway/usr/share/sway/scripts/sunset.sh]:
```suggestion
			longitude="${longitude:-65}"
```

File: community/sway/usr/share/sway/scripts/sunset.sh [hunk 3]
```diff
@@ -1,5 +1,58 @@
...
+			latitude='60'
```
Comment (boredland) [inline community/sway/usr/share/sway/scripts/sunset.sh]:
```suggestion
			latitude="${latitude:-60}"
```

File: community/sway/usr/share/sway/scripts/sunset.sh [hunk 4]
```diff
@@ -1,5 +1,58 @@
...
\ No newline at end of file
+
+#Startup function
+function start(){
+	source $HOME/.config/wlsunset/config
+
+	if [ ${location} = "on" ]; 
+	then
+		CONTENT=$(curl -s https://freegeoip.app/json/)
+		longitude=$(echo $CONTENT | jq .longitude)
```
Comment (boredland) [inline community/sway/usr/share/sway/scripts/sunset.sh]:
```suggestion
		longitude=${$(echo $CONTENT | jq '.longitude // empty'):-${longitude:-65}}
```

File: community/sway/usr/share/sway/scripts/sunset.sh [hunk 5]
```diff
@@ -1,5 +1,58 @@
...
+		latitude=$(echo $CONTENT | jq .latitude)
```
Comment (boredland) [inline community/sway/usr/share/sway/scripts/sunset.sh]:
```suggestion
		latitude=${$(echo $CONTENT | jq '.latitude // empty'):-${latitude:-65}}
```

File: community/sway/usr/share/sway/scripts/sunset.sh [hunk 6]
```diff
@@ -1,5 +1,58 @@
...
+		if [ -e $longitude ];
```
Comment (boredland) [inline community/sway/usr/share/sway/scripts/sunset.sh]:
```suggestion
```

File: community/sway/usr/share/sway/scripts/sunset.sh [hunk 7]
```diff
@@ -1,5 +1,58 @@
...
+		then
```
Comment (boredland) [inline community/sway/usr/share/sway/scripts/sunset.sh]:
```suggestion
```

File: community/sway/usr/share/sway/scripts/sunset.sh [hunk 8]
```diff
@@ -1,5 +1,58 @@
...
+			echo location ERROR: freegeoip.app
```
Comment (boredland) [inline community/sway/usr/share/sway/scripts/sunset.sh]:
```suggestion
```

File: community/sway/usr/share/sway/scripts/sunset.sh [hunk 9]
```diff
@@ -1,5 +1,58 @@
...
+			longitude="${longitude:-65}"
```
Comment (boredland) [inline community/sway/usr/share/sway/scripts/sunset.sh]:
```suggestion
```

File: community/sway/usr/share/sway/scripts/sunset.sh [hunk 10]
```diff
@@ -1,5 +1,58 @@
...
+			latitude="${latitude:-60}"
```
Comment (boredland) [inline community/sway/usr/share/sway/scripts/sunset.sh]:
```suggestion
```

File: community/sway/usr/share/sway/scripts/sunset.sh [hunk 11]
```diff
@@ -1,5 +1,58 @@
...
+		else
```
Comment (boredland) [inline community/sway/usr/share/sway/scripts/sunset.sh]:
```suggestion
```

File: community/sway/usr/share/sway/scripts/sunset.sh [hunk 12]
```diff
@@ -1,5 +1,58 @@
...
+			echo location OK:  $latitude $longitude
```
Comment (boredland) [inline community/sway/usr/share/sway/scripts/sunset.sh]:
```suggestion
```

File: community/sway/usr/share/sway/scripts/sunset.sh [hunk 13]
```diff
@@ -1,5 +1,58 @@
...
+		fi
```
Comment (boredland) [inline community/sway/usr/share/sway/scripts/sunset.sh]:
```suggestion
```

**Comment by boredland** [comment]
> I don't run sunset but I made all the changes and it doesn't show on the waybar so the exec-if works. Now I noticed we need a config file now which I believe is to customize it better now correct?

@Mathew-D your comments raised some questions/ideas on my side about how much sense that config file really makes. so I went and put the defaults inside the script which makes the overall ux a lot better. basically you now can provide a config file with 
```
location=on
```
to enable geoip. the default is to use just time.

**Comment by Mathew-D** [comment]
Shouldn't the: [[ -f "$HOME/.config/wlsunset/config" ]] && source "$HOME/.config/wlsunset/config" be after the defaults.  That way it overrides the defaults in the script if it exists.  As right now as I read it it sees if the file is there then loads it.  Afterwards it overrides all the info with the defaults in the script

**Comment by boredland** [comment]
> Shouldn't the: [[ -f "$HOME/.config/wlsunset/config" ]] && source "$HOME/.config/wlsunset/config" be after the defaults. That way it overrides the defaults in the script if it exists. As right now as I read it it sees if the file is there then loads it. Afterwards it overrides all the info with the defaults in the script

no, thats bash greatness:
```bash
${temp_low:-"4000"}
```
means:
return temp_low if exists, fall back to 4000.

see
image

so IMO the order wouldn't change anything and the current order has the benefit that if for some reason sb does set 
```bash
temp_low=
```
in his config, this will be overridden.

**Comment by Mathew-D** [comment]
I didn't know that.  thank you

**Pull request merged**